src/main/java/com/nicoislost/ZoomO.java

```diff
@@ -28,197 +45,160 @@ public void onInitializeClient() {
-						break;
-					}
-					if (found) {
-						zoomOrder[i] = zoomOrder[i + 1];
-					}
-				}
-				return zoomOrder[0];
-			}
-		} else { // All three keys are pressed.
-			if (zoomOrder[2] == 0) {
-				// First time pressing the third key.
-				try {
-					zoomOrder[2] = 6 / (zoomOrder[0] * zoomOrder[1]); //TODO find a better way to avoid dividing by zero.
-				} catch (Exception e) {
-					zoomOrder[2] = 0;
-				}
-			}
-			return zoomOrder[0];
+	/**
+	 * @return Returns the currently active key binding or null if none are active
+	 */
+	private static KeyBind getCurrentKeyBind() {
+		if (!isZooming() || ACTIVE_KEYS.isEmpty()) {
+			return null;
 		}
-		return 0;
+		return ACTIVE_KEYS.get(0);
 	}
 
 	/**
-	 * Translates the number output from {@link #zoomModifierNum()} to a double with
-	 * the zoom amount defined in the config.
-	 * 
-	 * @param keyNum - 1, 2, or 3
-	 * @return A double that can be interpreted as a FOV value in the source based
-	 *         on corresponding values in the config. Returns 0 if keyNum is
-	 *         invalid.
+	 * Returns the configured zoom modifier for the currently active key binding
+	 * @return The configure zoom modifier value for the active key binding or 0 if no key binding is active
 	 */
-	public static double ZoomModifier(int keyNum) {
-		if (keyNum == 1) {
-			return CONFIG.Zoom1();
-		} else if (keyNum == 2) {
-			return CONFIG.Zoom2();
-		} else if (keyNum == 3) {
-			return CONFIG.Zoom3();
+	private static double getCurrentZoomModifier() {
+		KeyBind keyBind = getCurrentKeyBind();
+		if (keyBind == null) {
+			return 0;
+		}
+		LOGGER.info("Current key = " + keyBind);
+		double modifier = 0;
+		// It would be nice to link the binding and the config directly, but small steps
+		switch (keyBind) {
+			case LEVEL_1 -> modifier = CONFIG.Zoom1();
+			case LEVEL_2 -> modifier = CONFIG.Zoom2();
+			case LEVEL_3 -> modifier = CONFIG.Zoom3();
 		}
-		return 0;
+		return modifier;
 	}
 
 	/**
-	 * Tells us how many keybindings are pressed to use in
-	 * {@link #zoomModifierNum()}
-	 * 
-	 * @return 0-3 depending on how many keys are pressed.
+	 * Delegates the functionality of the MixIn callback
+	 * @param callbackInfo
 	 */
-	public static int keysPressed() {
-		//This converts our booleans into ints and sums them up.
-		return (key1() ? 1 : 0) + (key2() ? 1 : 0) + (key3() ? 1 : 0);
+	// One thing to remember is, this is going to be called ALOT while one of the key bindings is being pressed,
+	// so we should be conscious of doing as little work as fast as possible - so, caching values if possible and/or
+	// reducing the amount of computational work we're doing.
+	// One thing we could do is check the previous state with the current state and if nothing's changed, returned
+	// the previously calculated value, but, arguably, comparing two `List` isn't the fastest thing in the world either,
+	// so maybe a `BitSet` instead?
+	public static void getZoomLevel(CallbackInfoReturnable<Double> callbackInfo) {
```

Comment:
Arguably, this should called something like `applyZoomLevel`, but I took the name from the `ZoomOMixIn` class.

Why?  Well, because it doesn't actually return anything, so you're not "getting" anything